Use callbag-from-iter under the hood

Andarist opened the pull request:
Let's face it `loop`, `inLoop`, `got1`, `ended` etc dance is quirky, hard to follow. IMHO it's much better to contain it in a single package (when possible) and just reuse it.

Andarist committed:
Use callbag-from-iter under the hood

franciscotln commented on index.js:
```diff
@@ -11,37 +13,19 @@ const checkArgs = (from, to, step) => {
 
```
I've noticed that it's possible to get into an infinite loop if someone enters a descending range with a positive step, so please add one more check inside the 'checkArgs' function
```js
if (from > to && step > 0) {
  throw new Error('Descending range must have negative step. Got +' + step);
}
```

franciscotln replied:
this should be at the end of the checkArgs function to be sure that all args are values :-)

franciscotln commented on index.js:
```diff
@@ -11,37 +13,19 @@ const checkArgs = (from, to, step) => {
-      }
-      inLoop = false;
-    };
+  let sent = from - step;
```
you can name this variable as `value` instead of `sent`

franciscotln commented on package.json:
```diff
@@ -1,7 +1,9 @@
   "description": "Callbag source that generates a range of numbers",
-  "dependencies": {},
+  "dependencies": {
+    "callbag-from-iter": "^1.2.0"
```
please use minor versioning `"~1.2.0"`

Andarist replied:
What's the reasoning behind this? It's rather not a common practice and might cause duplicated dependencies in people's applications.

franciscotln replied:
Ok, I agree with you. On one hand I don't want to have dependencies breaking the behaviour of our implementation, on the other hand it's not good to duplicate dependencies. You can leave it as is.

Andarist replied:
minor versions shouldnt break the backward compatibility, so it should be "safe" to use ^ instead of ~, but ofc they might because semver versions are bumped by humans (although the same argument applies to ~ - somebody might release a breaking change in patch version)

franciscotln commented:
@Andarist there are also some conflicts in index.js :-)

Andarist committed:
Hide checkArgs behind NODE_ENV check

Andarist committed:
Distribute module entry

Andarist commented:
I've fixed the conflict & adjusted the code according to your requests (mostly).

I've also added some additional stuff (latest 2 commits) as part of this PR because I didnt want to have guaranteed conflicts later. Let me know what do you think.

Andarist commented on index.js:
```diff
@@ -7,42 +9,31 @@ const checkArgs = (from, to, step) => {
+  }
 };
 
+const isProd = process.env.NODE_ENV === 'production'
```
this should allow dropping the checkArgs in production mode when bundling this library for usage on the web

franciscotln replied:
yeah, good point
